File: mq/rabbitmq/py/topic/publisher.py
#!/usr/bin/env python3
#
# Copyright (C) 2019 jayson
#
import json
import random
import utils.logging
from utils.connection import get_exchange_connection
from utils.random import random_string, random_sleep
from config import RABBITMQ_MASTER


if __name__ == '__main__':
    with get_exchange_connection(RABBITMQ_MASTER, 'logs_topic', 'topic') as channel:
        name = random_string(8)
        for index in range(2 ** 32):
            module = random.choice(['account', 'base', 'statistics', 'open'])
            level = random.choice(['debug', 'info', 'warning', 'error'])
            message = json.dumps({'producer': name, 'index': index, 'from': 'python', 'body': random_string(32)})
            channel.basic_publish('logs_topic', routing_key=module + '.' + level, body=message.encode('utf-8'))
            utils.logging.info("send message<{} | {:^20} | {}> to logs_routing from python".format(
                name, module + '.' + level, index))
            random_sleep(2)

File: mq/rabbitmq/php/src/topic/publisher.php
<?php
/**
 * rabbitmq hello-world
 */
require_once __DIR__ . '/../bootstrap.php';


$connection = new \PhpAmqpLib\Connection\AMQPStreamConnection(RABBITMQ_MASTER, 5672, 'guest', 'guest');
$channel = $connection->channel();

$channel->exchange_declare('logs_topic', 'topic', false, true, false);

$producer = bin2hex(random_bytes(4));
$levels = ['debug', 'info', 'warning', 'error'];
$modules = ['account', 'base', 'statistics', 'open'];
for ($index = 0; $index < 10000; ++$index) {
    $channel->basic_publish(new \PhpAmqpLib\Message\AMQPMessage(json_encode([
        'from' => 'php',
        'producer' => $producer,
        'index' => $index,
        'body' => bin2hex(random_bytes(16))
    ])), 'logs_topic', $modules[random_int(0, 3)] . '. ' . $levels[random_int(0, 3)]);
    sleep(random_int(1, 2));
}
